numba/targets/arrayobj.py

```diff
@@ -905,6 +905,44 @@ def array_argmax_impl(arry):
     return context.compile_internal(builder, array_argmax_impl, sig, args)
 
 
+@builtin
+@implement(numpy.median, types.Kind(types.Array))
+@implement("array.mean", types.Kind(types.Array))
+def array_median(context, builder, sig, args):
+    # XXX: the implementation doesn't work for the case if array is not one
+    # dimentional
+    def partition(A, p, r):
+        # TODO: make it randomized
+        x = A[r]
```

Comment:
This will make the algorithm O(n**2) if the array is already sorted. You could first swap `A[r]` and `A[(p + r) // 2]` to avoid that.